lib/ui/pages/score/score_page.dart

```diff
@@ -125,188 +142,160 @@ class _ScoreViewerPageState extends State<ScoreViewerPage> with TickerProviderSt
+    tabLabelList.clear();
+    tabChildList.clear();
+
     try {
       if (courseScoreCredit.graduationInformation.isSelect) {
         tabLabelList.add(_buildTabLabel(R.current.creditSummary));
-        tabChildList.add(StatefulBuilder(
-          builder: (BuildContext context, void Function(void Function()) setState) {
-            return AnimationLimiter(
-              child: Column(
-                children: AnimationConfiguration.toStaggeredList(
-                  childAnimationBuilder: (widget) => SlideAnimation(
-                    verticalOffset: 50.0,
-                    child: FadeInAnimation(child: widget),
-                  ),
-                  children: <Widget>[
-                    _buildSummary(),
-                    _buildGeneralLessonItem(),
-                    _buildOtherDepartmentItem(),
-                    _buildWarning(),
-                  ],
+        tabChildList.add(
+          AnimationLimiter(
+            child: Column(
+              children: AnimationConfiguration.toStaggeredList(
+                childAnimationBuilder: (widget) => SlideAnimation(
+                  verticalOffset: 50.0,
+                  child: FadeInAnimation(child: widget),
                 ),
+                children: [
+                  _buildSummary(),
+                  _buildGeneralLessonItem(),
+                  _buildOtherDepartmentItem(),
+                  const ScoreCalculationWarning(),
+                ],
               ),
-            );
-          },
-        ));
+            ),
+          ),
+        );
       }
     } catch (e, stack) {
       Log.eWithStack(e.toString(), stack);
     }
+
     for (int i = 0; i < courseScoreList.length; i++) {
-      SemesterCourseScoreJson courseScore = courseScoreList[i];
+      final courseScore = courseScoreList[i];
       tabLabelList.add(_buildTabLabel("${courseScore.semester.year}-${courseScore.semester.semester}"));
       tabChildList.add(_buildSemesterScores(courseScore));
     }
+
     if (_tabController != null) {
-      if (tabChildList.length != _tabController.length) {
-        _tabController.dispose();
+      if (tabChildList.length != _tabController?.length) {
+        _tabController?.dispose();
         _tabController = TabController(length: tabChildList.length, vsync: this);
       }
     } else {
       _tabController = TabController(length: tabChildList.length, vsync: this);
     }
+
     _currentTabIndex = 0;
-    _tabController.animateTo(_currentTabIndex);
+    _tabController?.animateTo(_currentTabIndex);
     setState(() {});
   }
 
-  Widget _buildTabLabel(String title) {
-    return Padding(
-      padding: const EdgeInsets.only(
-        left: 12,
-        right: 12,
-      ),
-      child: Tab(
-        text: title,
-      ),
-    );
-  }
+  Widget _buildTabLabel(String title) => Padding(
+        padding: const EdgeInsets.only(
+          left: 12,
+          right: 12,
+        ),
+        child: Tab(
+          text: title,
+        ),
```

Comment:
[nit]
Just my personal preference, if the content inside a widget is short enough I would prefer to inline it.

```suggestion
        padding: const EdgeInsets.only(left: 12, right: 12),
        child: Tab(text: title),
```